src/main/java/com/scaler/lld/design/assignments/singleton/FileBasedConfigurationManagerImpl.java

```diff
@@ -5,46 +5,69 @@ public class FileBasedConfigurationManagerImpl extends FileBasedConfigurationMan
     @Override
     public String getConfiguration(String key) {
         // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+        return properties.getProperty(key);
     }
 
     @Override
     public <T> T getConfiguration(String key, Class<T> type) {
         // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+        if(INSTANCE!=null){
+            String str=INSTANCE.properties.getProperty(key);
+            if(str!=null)
+                return convert(str,type);
+            else
+                return null;
```

Comment:
1. Always add parentheses when starting a block like if, else, while, etc. Tomorrow, someone can come in and add code above/below missing that there are no brackets. 
2. [MINOR] This implementation works. You can improve it by using the `Optional` class to reduce the level of indentation.